diff -rupN devicetree.orig/pcw.dtsi devicetree/pcw.dtsi
--- devicetree.orig/pcw.dtsi	2020-07-01 16:52:50.025692220 +0100
+++ devicetree/pcw.dtsi	2020-07-01 16:49:36.487202058 +0100
@@ -47,6 +47,11 @@
 	is-decoded-cs = <0>;
 	num-cs = <3>;
 	status = "okay";
+	spidev@0 {
+		compatible = "spidev";
+		reg = <0x0>;
+		spi-max-frequency = <10000000>;
+	};
 };
 &uart1 {
 	cts-override ;
@@ -55,7 +60,7 @@
 	status = "okay";
 };
 &usb0 {
-	phy_type = "ulpi";
+	usb-phy = <&usb_phy0>;
 	status = "okay";
 	usb-reset = <&gpio0 7 0>;
 };
diff -rupN devicetree.orig/system-top.dts devicetree/system-top.dts
--- devicetree.orig/system-top.dts	2020-07-01 16:52:50.025692220 +0100
+++ devicetree/system-top.dts	2020-07-01 16:49:36.487202058 +0100
@@ -18,10 +18,17 @@
 		i2c0 = &i2c0;
 		serial0 = &uart1;
 		spi0 = &qspi;
-		spi1 = &spi0;
+		spi2 = &spi0;
+	};
+	usb_phy0: phy0 {
+		#phy-cells = <0>;
+		compatible = "ulpi-phy";
+		reg = <0xe0002000 0x1000>;
+		view-port = <0x0170>;
+		drv-vbus;
 	};
 	memory {
 		device_type = "memory";
-		reg = <0x0 0x20000000>;
+		reg = <0x0 0x10000000>;
 	};
 };
